static/js/rendered_markdown.js

```diff
@@ -132,7 +132,11 @@ export const update_elements = (content) => {
                 // If the stream has been deleted,
                 // stream_data.maybe_get_stream_name might return
                 // undefined.  Otherwise, display the current stream name.
-                $(this).text("#" + stream_name);
+                let stream_icon = "#";
+                if (stream_data.get_invite_only(stream_name)) {
+                    stream_icon = "<i class='fa fa-lock' aria-hidden='true'></i>";
```

Comment:
Yup, it's not advisable to interpolates unescaped text into HTML.
```suggestion
                    stream_icon = $("<i>", {class: "fa fa-lock"});
```